projects/netgrif-components-core/src/lib/side-menu/content-components/user-assign/base-user-assign-list/abstract-base-user-assign-list.component.ts

```diff
@@ -0,0 +1,62 @@
+import {Component, Input, OnDestroy, OnInit, ViewChild} from "@angular/core";
+import {RolesObject, UserListItem, UserListService} from "../../../../user/services/user-list.service";
+import {FormControl} from "@angular/forms";
+import {CdkVirtualScrollViewport} from "@angular/cdk/scrolling";
+import {debounceTime} from "rxjs/operators";
+import {UserValue} from "../../../../data-fields/user-field/models/user-value";
+
+@Component({
+    selector: 'ncc-base-user-assign-list',
+    template: ''
+})
+export abstract class AbstractBaseUserAssignListComponent implements OnInit, OnDestroy {
+
+    protected SEARCH_DEBOUNCE_TIME = 600;
+
+    @Input() searchUserControl: FormControl;
+
+    @Input() roles: RolesObject | Array<string>;
+
+    @Input() negativeRoles: RolesObject | Array<string>;
+
+    @ViewChild(CdkVirtualScrollViewport) public viewport: CdkVirtualScrollViewport;
+
+    constructor(protected _userListService: UserListService) {
+
+    }
+
+    ngOnInit() {
+        this.searchUserControl.valueChanges.pipe(debounceTime(this.SEARCH_DEBOUNCE_TIME)).subscribe(searchText => {
+            this._userListService.reload(searchText);
+        });
+        if (this.roles instanceof Array) {
```

Comment:
why is the Array check done here with `instanceof` and in the `ContainsPipe` with `Array.isArray()`?

Reply:
As I mentioned earlier, I just copied this code to create an abstraction layer above two components `AbstractUserAssignListComponent` and `AbstractMultiUserAssignListComponent`, this code was present in `AbstractUserAssignListComponent` since 2020, so I do not changed the functioning part, if you wish, I can refactor then.